#Capstone Project Google Data Analytics


Case Study 2: How Can a Wellness Technology Company Play It Smart?

## 1. Ask


*   Identify the business task : The business task is to analyze smart device data to gain insight into how consumers are using their smart devices. The insights will be used to guide marketing strategy for Bellabeat
*   Key Stakeholders: Urska Srsen (Co-founder and Chief Creative Officer), Sando Mur (Mathematician and Cofounder), Marketing Analytics Team






## 2.   Prepare



*   Data stored: The data source is from Kaggle and is freely accessible. I stored my data on a folder made for the project purpose.
*   Data organization: The data is organized as long format, since every user have it's own ID which is used repetitively in some columns. 
*   Credibility and Bias: 
<br> The reliability of the data can be confirmed by the numbers of users that the data includes. 
<br> However, the data is not original because it is a public data coming from a second party. 
<br> The data do contains information about what I am going to discover, but it still lacks of information as it only displays 30 users. Thus, the data is not comprehensive. 
<br> The data is taken from 2016, which is 6 years ago and I considered it not current. 
*   Data Integrity: To verify the data integrity, I need to jump into the process phase. Here, I discover that: 
<br> The column 'LoggedActivitiesDistance' in dailyActivity data have 0 values and do not inform anything. 
<br> In addition, the 'Fat' column in the weightLogInfoData has 65 null values.    
<br> I checked the consintencies of user counts in every data. Most of them contains 33 uniqued ids, but 4 dataset does have inconsintencies, which are heartrate_seconds (14), minuteSleep (24), sleepDay (24), and weightLogInfo (8). 








## 3. Process

In [ ]:
import os
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns

In [ ]:
os.environ['KAGGLE_USERNAME'] = "izzazzhr"
os.environ['KAGGLE_KEY'] = "1705a179e0f81c936f8e5b8b30878aa4"

In [ ]:
!kaggle datasets download -d arashnic/fitbit

 21% 5.00M/24.1M [00:00<00:00, 47.8MB/s]
100% 24.1M/24.1M [00:00<00:00, 125MB/s] 


In [ ]:
!unzip -q /content/fitbit.zip

###Load Data

In [ ]:
dailyActivity = pd.read_csv("/content/Fitabase Data 4.12.16-5.12.16/dailyActivity_merged.csv") 

In [ ]:
dailyCalories = pd.read_csv("/content/Fitabase Data 4.12.16-5.12.16/dailyCalories_merged.csv")

In [ ]:
dailyIntensities = pd.read_csv("/content/Fitabase Data 4.12.16-5.12.16/dailyIntensities_merged.csv")

In [ ]:
dailySteps = pd.read_csv("/content/Fitabase Data 4.12.16-5.12.16/dailySteps_merged.csv")

In [ ]:
heartrate_seconds=pd.read_csv("/content/Fitabase Data 4.12.16-5.12.16/heartrate_seconds_merged.csv")

In [ ]:
hourlyCalories = pd.read_csv("/content/Fitabase Data 4.12.16-5.12.16/hourlyCalories_merged.csv")

In [ ]:
hourlyIntensities = pd.read_csv("/content/Fitabase Data 4.12.16-5.12.16/hourlyIntensities_merged.csv")

In [ ]:
hourlySteps = pd.read_csv("/content/Fitabase Data 4.12.16-5.12.16/hourlySteps_merged.csv")

In [ ]:
minuteCaloriesNarrow = pd.read_csv("/content/Fitabase Data 4.12.16-5.12.16/minuteCaloriesNarrow_merged.csv")

In [ ]:
minuteCaloriesWide = pd.read_csv("/content/Fitabase Data 4.12.16-5.12.16/minuteCaloriesWide_merged.csv")

In [ ]:
minuteIntensitiesNarrow = pd.read_csv("/content/Fitabase Data 4.12.16-5.12.16/minuteIntensitiesNarrow_merged.csv")

In [ ]:
minuteIntensitiesWide = pd.read_csv("/content/Fitabase Data 4.12.16-5.12.16/minuteIntensitiesWide_merged.csv")

In [ ]:
minuteMETsNarrow = pd.read_csv("/content/Fitabase Data 4.12.16-5.12.16/minuteMETsNarrow_merged.csv")

In [ ]:
minuteSleep = pd.read_csv("/content/Fitabase Data 4.12.16-5.12.16/minuteSleep_merged.csv")

In [ ]:
minuteStepsNarrow = pd.read_csv("/content/Fitabase Data 4.12.16-5.12.16/minuteStepsNarrow_merged.csv")

In [ ]:
minuteStepsWide = pd.read_csv("/content/Fitabase Data 4.12.16-5.12.16/minuteStepsWide_merged.csv")

In [ ]:
sleepDay = pd.read_csv("/content/Fitabase Data 4.12.16-5.12.16/sleepDay_merged.csv")

In [ ]:
weightLogInfo = pd.read_csv("/content/Fitabase Data 4.12.16-5.12.16/weightLogInfo_merged.csv")

### Data Cleaning

####Daily Activity

In [ ]:
dailyActivity.head()

Id ActivityDate  TotalSteps  TotalDistance  TrackerDistance  \
0  1503960366    4/12/2016       13162           8.50             8.50   
1  1503960366    4/13/2016       10735           6.97             6.97   
2  1503960366    4/14/2016       10460           6.74             6.74   
3  1503960366    4/15/2016        9762           6.28             6.28   
4  1503960366    4/16/2016       12669           8.16             8.16   

   LoggedActivitiesDistance  VeryActiveDistance  ModeratelyActiveDistance  \
0                       0.0                1.88                      0.55   
1                       0.0                1.57                      0.69   
2                       0.0                2.44                      0.40   
3                       0.0                2.14                      1.26   
4                       0.0                2.71                      0.41   

   LightActiveDistance  SedentaryActiveDistance  VeryActiveMinutes  \
0                 6.06                      0.0                 25   
1                 4.71                      0.0                 21   
2                 3.91                      0.0                 30   
3                 2.83                      0.0                 29   
4                 5.04                      0.0                 36   

   FairlyActiveMinutes  LightlyActiveMinutes  SedentaryMinutes  Calories  
0                   13                   328               728      1985  
1                   19                   217               776      1797  
2                   11                   181              1218      1776  
3                   34                   209               726      1745  
4                   10                   221               773      1863

In [ ]:
dailyActivity = dailyActivity.drop(['LoggedActivitiesDistance'],axis=1)

In [ ]:
dailyActivity.isnull().sum()

Id                          0
ActivityDate                0
TotalSteps                  0
TotalDistance               0
TrackerDistance             0
VeryActiveDistance          0
ModeratelyActiveDistance    0
LightActiveDistance         0
SedentaryActiveDistance     0
VeryActiveMinutes           0
FairlyActiveMinutes         0
LightlyActiveMinutes        0
SedentaryMinutes            0
Calories                    0
dtype: int64

In [ ]:
dailyActivity['Id'].nunique()

33

#### Daily Calories

In [ ]:
dailyCalories.head()

Id ActivityDay  Calories
0  1503960366   4/12/2016      1985
1  1503960366   4/13/2016      1797
2  1503960366   4/14/2016      1776
3  1503960366   4/15/2016      1745
4  1503960366   4/16/2016      1863

In [ ]:
dailyCalories.isnull().sum()

Id             0
ActivityDay    0
Calories       0
dtype: int64

In [ ]:
dailyCalories['Id'].nunique()

33

#### Daily Intensities

In [ ]:
dailyIntensities.head()

Id ActivityDay  SedentaryMinutes  LightlyActiveMinutes  \
0  1503960366   4/12/2016               728                   328   
1  1503960366   4/13/2016               776                   217   
2  1503960366   4/14/2016              1218                   181   
3  1503960366   4/15/2016               726                   209   
4  1503960366   4/16/2016               773                   221   

   FairlyActiveMinutes  VeryActiveMinutes  SedentaryActiveDistance  \
0                   13                 25                      0.0   
1                   19                 21                      0.0   
2                   11                 30                      0.0   
3                   34                 29                      0.0   
4                   10                 36                      0.0   

   LightActiveDistance  ModeratelyActiveDistance  VeryActiveDistance  
0                 6.06                      0.55                1.88  
1                 4.71                      0.69                1.57  
2                 3.91                      0.40                2.44  
3                 2.83                      1.26                2.14  
4                 5.04                      0.41                2.71

In [ ]:
dailyIntensities.isnull().sum()

Id                          0
ActivityDay                 0
SedentaryMinutes            0
LightlyActiveMinutes        0
FairlyActiveMinutes         0
VeryActiveMinutes           0
SedentaryActiveDistance     0
LightActiveDistance         0
ModeratelyActiveDistance    0
VeryActiveDistance          0
dtype: int64

In [ ]:
dailyIntensities['Id'].nunique()

33

#### Daily Steps

In [ ]:
dailySteps.head()

Id ActivityDay  StepTotal
0  1503960366   4/12/2016      13162
1  1503960366   4/13/2016      10735
2  1503960366   4/14/2016      10460
3  1503960366   4/15/2016       9762
4  1503960366   4/16/2016      12669

In [ ]:
dailySteps.isnull().sum()

Id             0
ActivityDay    0
StepTotal      0
dtype: int64

In [ ]:
dailySteps['Id'].nunique()

33

#### Heartrate Seconds

In [ ]:
heartrate_seconds.head()

Id                  Time  Value
0  2022484408  4/12/2016 7:21:00 AM     97
1  2022484408  4/12/2016 7:21:05 AM    102
2  2022484408  4/12/2016 7:21:10 AM    105
3  2022484408  4/12/2016 7:21:20 AM    103
4  2022484408  4/12/2016 7:21:25 AM    101

In [ ]:
heartrate_seconds.isnull().sum()

Id       0
Time     0
Value    0
dtype: int64

In [ ]:
heartrate_seconds['Id'].nunique()

14

#### Hourly Calories

In [ ]:
hourlyCalories.head()

Id           ActivityHour  Calories
0  1503960366  4/12/2016 12:00:00 AM        81
1  1503960366   4/12/2016 1:00:00 AM        61
2  1503960366   4/12/2016 2:00:00 AM        59
3  1503960366   4/12/2016 3:00:00 AM        47
4  1503960366   4/12/2016 4:00:00 AM        48

In [ ]:
hourlyCalories.isnull().sum()

Id              0
ActivityHour    0
Calories        0
dtype: int64

In [ ]:
hourlyCalories['Id'].nunique()

33

#### Hourly Intensities

In [ ]:
hourlyIntensities.head()

Id           ActivityHour  TotalIntensity  AverageIntensity
0  1503960366  4/12/2016 12:00:00 AM              20          0.333333
1  1503960366   4/12/2016 1:00:00 AM               8          0.133333
2  1503960366   4/12/2016 2:00:00 AM               7          0.116667
3  1503960366   4/12/2016 3:00:00 AM               0          0.000000
4  1503960366   4/12/2016 4:00:00 AM               0          0.000000

In [ ]:
hourlyIntensities.isnull().sum()

Id                  0
ActivityHour        0
TotalIntensity      0
AverageIntensity    0
dtype: int64

In [ ]:
hourlyIntensities['Id'].nunique()

33

#### Hourly Steps

In [ ]:
hourlySteps.head()

Id           ActivityHour  StepTotal
0  1503960366  4/12/2016 12:00:00 AM        373
1  1503960366   4/12/2016 1:00:00 AM        160
2  1503960366   4/12/2016 2:00:00 AM        151
3  1503960366   4/12/2016 3:00:00 AM          0
4  1503960366   4/12/2016 4:00:00 AM          0

In [ ]:
hourlySteps.isnull().sum()

Id              0
ActivityHour    0
StepTotal       0
dtype: int64

In [ ]:
hourlySteps['Id'].nunique()

33

#### Minute Calories Narrow

In [ ]:
minuteCaloriesNarrow.head()

Id         ActivityMinute  Calories
0  1503960366  4/12/2016 12:00:00 AM    0.7865
1  1503960366  4/12/2016 12:01:00 AM    0.7865
2  1503960366  4/12/2016 12:02:00 AM    0.7865
3  1503960366  4/12/2016 12:03:00 AM    0.7865
4  1503960366  4/12/2016 12:04:00 AM    0.7865

In [ ]:
minuteCaloriesNarrow.isnull().sum()

Id                0
ActivityMinute    0
Calories          0
dtype: int64

In [ ]:
minuteCaloriesNarrow['Id'].nunique()

33

#### Minute Calories Wide

In [ ]:
minuteCaloriesWide.head()

Id           ActivityHour  Calories00  Calories01  Calories02  \
0  1503960366  4/13/2016 12:00:00 AM      1.8876      2.2022      0.9438   
1  1503960366   4/13/2016 1:00:00 AM      0.7865      0.7865      0.7865   
2  1503960366   4/13/2016 2:00:00 AM      0.7865      0.7865      0.7865   
3  1503960366   4/13/2016 3:00:00 AM      0.7865      0.7865      0.7865   
4  1503960366   4/13/2016 4:00:00 AM      0.7865      0.7865      0.7865   

   Calories03  Calories04  Calories05  Calories06  Calories07  ...  \
0      0.9438      0.9438      2.0449      0.9438      2.2022  ...   
1      0.7865      0.9438      0.9438      0.9438      0.7865  ...   
2      0.7865      0.7865      0.7865      0.7865      0.7865  ...   
3      0.7865      0.7865      0.7865      0.7865      0.7865  ...   
4      0.7865      0.7865      0.7865      0.7865      0.7865  ...   

   Calories50  Calories51  Calories52  Calories53  Calories54  Calories55  \
0      0.9438      2.0449      2.0449      0.9438      2.3595      1.8876   
1      0.7865      0.7865      0.7865      0.7865      0.7865      0.7865   
2      0.7865      0.7865      0.7865      0.7865      0.7865      0.7865   
3      0.7865      0.7865      0.7865      0.7865      0.7865      0.7865   
4      0.7865      0.7865      0.7865      0.7865      0.7865      0.7865   

   Calories56  Calories57  Calories58  Calories59  
0      0.9438      0.9438      0.9438      0.9438  
1      0.7865      0.7865      0.7865      0.7865  
2      0.7865      0.7865      0.7865      0.7865  
3      0.7865      0.7865      0.7865      0.7865  
4      0.7865      0.7865      0.7865      0.7865  

[5 rows x 62 columns]

In [ ]:
minuteCaloriesWide.head().isnull().sum()

Id              0
ActivityHour    0
Calories00      0
Calories01      0
Calories02      0
               ..
Calories55      0
Calories56      0
Calories57      0
Calories58      0
Calories59      0
Length: 62, dtype: int64

In [ ]:
minuteCaloriesWide['Id'].nunique()

33

#### Minute Intensities Narrows

In [ ]:
minuteIntensitiesNarrow.head()

Id         ActivityMinute  Intensity
0  1503960366  4/12/2016 12:00:00 AM          0
1  1503960366  4/12/2016 12:01:00 AM          0
2  1503960366  4/12/2016 12:02:00 AM          0
3  1503960366  4/12/2016 12:03:00 AM          0
4  1503960366  4/12/2016 12:04:00 AM          0

In [ ]:
minuteIntensitiesNarrow.isnull().sum()

Id                0
ActivityMinute    0
Intensity         0
dtype: int64

In [ ]:
minuteIntensitiesNarrow['Id'].nunique()

33

#### Minute Intensities Wide

In [ ]:
minuteIntensitiesWide.head()

Id           ActivityHour  Intensity00  Intensity01  Intensity02  \
0  1503960366  4/13/2016 12:00:00 AM            1            1            0   
1  1503960366   4/13/2016 1:00:00 AM            0            0            0   
2  1503960366   4/13/2016 2:00:00 AM            0            0            0   
3  1503960366   4/13/2016 3:00:00 AM            0            0            0   
4  1503960366   4/13/2016 4:00:00 AM            0            0            0   

   Intensity03  Intensity04  Intensity05  Intensity06  Intensity07  ...  \
0            0            0            1            0            1  ...   
1            0            0            0            0            0  ...   
2            0            0            0            0            0  ...   
3            0            0            0            0            0  ...   
4            0            0            0            0            0  ...   

   Intensity50  Intensity51  Intensity52  Intensity53  Intensity54  \
0            0            1            1            0            1   
1            0            0            0            0            0   
2            0            0            0            0            0   
3            0            0            0            0            0   
4            0            0            0            0            0   

   Intensity55  Intensity56  Intensity57  Intensity58  Intensity59  
0            1            0            0            0            0  
1            0            0            0            0            0  
2            0            0            0            0            0  
3            0            0            0            0            0  
4            0            0            0            0            0  

[5 rows x 62 columns]

In [ ]:
minuteIntensitiesWide.head().isnull().sum()

Id              0
ActivityHour    0
Intensity00     0
Intensity01     0
Intensity02     0
               ..
Intensity55     0
Intensity56     0
Intensity57     0
Intensity58     0
Intensity59     0
Length: 62, dtype: int64

In [ ]:
minuteIntensitiesWide['Id'].nunique()

33

#### Minute METs Narrow

In [ ]:
minuteMETsNarrow.head()

Id         ActivityMinute  METs
0  1503960366  4/12/2016 12:00:00 AM    10
1  1503960366  4/12/2016 12:01:00 AM    10
2  1503960366  4/12/2016 12:02:00 AM    10
3  1503960366  4/12/2016 12:03:00 AM    10
4  1503960366  4/12/2016 12:04:00 AM    10

In [ ]:
minuteMETsNarrow.head().isnull().sum()

Id                0
ActivityMinute    0
METs              0
dtype: int64

In [ ]:
minuteMETsNarrow['Id'].nunique()

33

#### Minute Sleep

In [ ]:
minuteSleep.head()

Id                  date  value        logId
0  1503960366  4/12/2016 2:47:30 AM      3  11380564589
1  1503960366  4/12/2016 2:48:30 AM      2  11380564589
2  1503960366  4/12/2016 2:49:30 AM      1  11380564589
3  1503960366  4/12/2016 2:50:30 AM      1  11380564589
4  1503960366  4/12/2016 2:51:30 AM      1  11380564589

In [ ]:
minuteSleep.isnull().sum()

Id       0
date     0
value    0
logId    0
dtype: int64

In [ ]:
minuteSleep['Id'].nunique()

24

#### Minute Steps Narrow

In [ ]:
minuteStepsNarrow.head(20)

Id         ActivityMinute  Steps
0   1503960366  4/12/2016 12:00:00 AM      0
1   1503960366  4/12/2016 12:01:00 AM      0
2   1503960366  4/12/2016 12:02:00 AM      0
3   1503960366  4/12/2016 12:03:00 AM      0
4   1503960366  4/12/2016 12:04:00 AM      0
5   1503960366  4/12/2016 12:05:00 AM      0
6   1503960366  4/12/2016 12:06:00 AM      0
7   1503960366  4/12/2016 12:07:00 AM      0
8   1503960366  4/12/2016 12:08:00 AM      0
9   1503960366  4/12/2016 12:09:00 AM      0
10  1503960366  4/12/2016 12:10:00 AM      0
11  1503960366  4/12/2016 12:11:00 AM      0
12  1503960366  4/12/2016 12:12:00 AM      0
13  1503960366  4/12/2016 12:13:00 AM      0
14  1503960366  4/12/2016 12:14:00 AM      0
15  1503960366  4/12/2016 12:15:00 AM      0
16  1503960366  4/12/2016 12:16:00 AM      0
17  1503960366  4/12/2016 12:17:00 AM      0
18  1503960366  4/12/2016 12:18:00 AM      0
19  1503960366  4/12/2016 12:19:00 AM      0

In [ ]:
minuteStepsNarrow.isnull().sum()

Id                0
ActivityMinute    0
Steps             0
dtype: int64

In [ ]:
minuteStepsNarrow['Id'].nunique()

33

#### Minute Steps Wide

In [ ]:
minuteStepsWide.head()

Id           ActivityHour  Steps00  Steps01  Steps02  Steps03  \
0  1503960366  4/13/2016 12:00:00 AM        4       16        0        0   
1  1503960366   4/13/2016 1:00:00 AM        0        0        0        0   
2  1503960366   4/13/2016 2:00:00 AM        0        0        0        0   
3  1503960366   4/13/2016 3:00:00 AM        0        0        0        0   
4  1503960366   4/13/2016 4:00:00 AM        0        0        0        0   

   Steps04  Steps05  Steps06  Steps07  ...  Steps50  Steps51  Steps52  \
0        0        9        0       17  ...        0        9        8   
1        0        0        0        0  ...        0        0        0   
2        0        0        0        0  ...        0        0        0   
3        0        0        0        0  ...        0        0        0   
4        0        0        0        0  ...        0        0        0   

   Steps53  Steps54  Steps55  Steps56  Steps57  Steps58  Steps59  
0        0       20        1        0        0        0        0  
1        0        0        0        0        0        0        0  
2        0        0        0        0        0        0        0  
3        0        0        0        0        0        0        0  
4        0        0        0        0        0        0        0  

[5 rows x 62 columns]

In [ ]:
minuteStepsWide.isnull().sum()

Id              0
ActivityHour    0
Steps00         0
Steps01         0
Steps02         0
               ..
Steps55         0
Steps56         0
Steps57         0
Steps58         0
Steps59         0
Length: 62, dtype: int64

In [ ]:
minuteStepsWide['Id'].nunique()

33

#### Sleep Day

In [ ]:
sleepDay.head()

Id               SleepDay  TotalSleepRecords  TotalMinutesAsleep  \
0  1503960366  4/12/2016 12:00:00 AM                  1                 327   
1  1503960366  4/13/2016 12:00:00 AM                  2                 384   
2  1503960366  4/15/2016 12:00:00 AM                  1                 412   
3  1503960366  4/16/2016 12:00:00 AM                  2                 340   
4  1503960366  4/17/2016 12:00:00 AM                  1                 700   

   TotalTimeInBed  
0             346  
1             407  
2             442  
3             367  
4             712

In [ ]:
sleepDay.isnull().sum()

Id                    0
SleepDay              0
TotalSleepRecords     0
TotalMinutesAsleep    0
TotalTimeInBed        0
dtype: int64

In [ ]:
sleepDay['Id'].nunique()

24

#### Weight Log Info

In [ ]:
weightLogInfo.head()

Id                   Date    WeightKg  WeightPounds   Fat  \
0  1503960366   5/2/2016 11:59:59 PM   52.599998    115.963147  22.0   
1  1503960366   5/3/2016 11:59:59 PM   52.599998    115.963147   NaN   
2  1927972279   4/13/2016 1:08:52 AM  133.500000    294.317120   NaN   
3  2873212765  4/21/2016 11:59:59 PM   56.700001    125.002104   NaN   
4  2873212765  5/12/2016 11:59:59 PM   57.299999    126.324875   NaN   

         BMI  IsManualReport          LogId  
0  22.650000            True  1462233599000  
1  22.650000            True  1462319999000  
2  47.540001           False  1460509732000  
3  21.450001            True  1461283199000  
4  21.690001            True  1463097599000

In [ ]:
weightLogInfo.isnull().sum()

Id                 0
Date               0
WeightKg           0
WeightPounds       0
Fat               65
BMI                0
IsManualReport     0
LogId              0
dtype: int64

In [ ]:
weightLogInfo = weightLogInfo.drop(['Fat'],axis=1)

In [ ]:
weightLogInfo['Id'].nunique()

8

## 4. Analyze

Upon analyzing the data, I found out that most of the data is merged to the other ones, hence, in this process I will only use:
<br> dailyActivity
<br> dailySteps
<br> sleepDay
<br> weightLogInfo



### Format Data

Before analyzing data, I need to format the date time for each data. The data type also need to be converted to datetime

In [ ]:
dailyActivity.head()

Id ActivityDate  TotalSteps  TotalDistance  TrackerDistance  \
0  1503960366    4/12/2016       13162           8.50             8.50   
1  1503960366    4/13/2016       10735           6.97             6.97   
2  1503960366    4/14/2016       10460           6.74             6.74   
3  1503960366    4/15/2016        9762           6.28             6.28   
4  1503960366    4/16/2016       12669           8.16             8.16   

   VeryActiveDistance  ModeratelyActiveDistance  LightActiveDistance  \
0                1.88                      0.55                 6.06   
1                1.57                      0.69                 4.71   
2                2.44                      0.40                 3.91   
3                2.14                      1.26                 2.83   
4                2.71                      0.41                 5.04   

   SedentaryActiveDistance  VeryActiveMinutes  FairlyActiveMinutes  \
0                      0.0                 25                   13   
1                      0.0                 21                   19   
2                      0.0                 30                   11   
3                      0.0                 29                   34   
4                      0.0                 36                   10   

   LightlyActiveMinutes  SedentaryMinutes  Calories  
0                   328               728      1985  
1                   217               776      1797  
2                   181              1218      1776  
3                   209               726      1745  
4                   221               773      1863

In [ ]:
dailyActivity.dtypes

Id                                   int64
ActivityDate                datetime64[ns]
TotalSteps                           int64
TotalDistance                      float64
TrackerDistance                    float64
VeryActiveDistance                 float64
ModeratelyActiveDistance           float64
LightActiveDistance                float64
SedentaryActiveDistance            float64
VeryActiveMinutes                    int64
FairlyActiveMinutes                  int64
LightlyActiveMinutes                 int64
SedentaryMinutes                     int64
Calories                             int64
year                               float64
date                               float64
month                              float64
dtype: object

In [ ]:
## Transform ActivityDate data type to datetime

dailyActivity['ActivityDate'] = pd.to_datetime(dailyActivity['ActivityDate'])

In [ ]:
dailySteps.head()

Id ActivityDay  StepTotal
0  1503960366   4/12/2016      13162
1  1503960366   4/13/2016      10735
2  1503960366   4/14/2016      10460
3  1503960366   4/15/2016       9762
4  1503960366   4/16/2016      12669

In [ ]:
dailySteps.dtypes

Id                      int64
ActivityDay    datetime64[ns]
StepTotal               int64
dtype: object

In [ ]:
dailySteps['ActivityDay'] = pd.to_datetime(dailySteps['ActivityDay'])

In [ ]:
sleepDay.head()

Id               SleepDay  TotalSleepRecords  TotalMinutesAsleep  \
0  1503960366  4/12/2016 12:00:00 AM                  1                 327   
1  1503960366  4/13/2016 12:00:00 AM                  2                 384   
2  1503960366  4/15/2016 12:00:00 AM                  1                 412   
3  1503960366  4/16/2016 12:00:00 AM                  2                 340   
4  1503960366  4/17/2016 12:00:00 AM                  1                 700   

   TotalTimeInBed  
0             346  
1             407  
2             442  
3             367  
4             712

In [ ]:
sleepDay.dtypes

Id                             int64
SleepDay              datetime64[ns]
TotalSleepRecords              int64
TotalMinutesAsleep             int64
TotalTimeInBed                 int64
dtype: object

In [ ]:
sleepDay['SleepDay'] = pd.to_datetime(sleepDay['SleepDay'])

In [ ]:
weightLogInfo.head()

Id                   Date    WeightKg  WeightPounds        BMI  \
0  1503960366   5/2/2016 11:59:59 PM   52.599998    115.963147  22.650000   
1  1503960366   5/3/2016 11:59:59 PM   52.599998    115.963147  22.650000   
2  1927972279   4/13/2016 1:08:52 AM  133.500000    294.317120  47.540001   
3  2873212765  4/21/2016 11:59:59 PM   56.700001    125.002104  21.450001   
4  2873212765  5/12/2016 11:59:59 PM   57.299999    126.324875  21.690001   

   IsManualReport          LogId  
0            True  1462233599000  
1            True  1462319999000  
2           False  1460509732000  
3            True  1461283199000  
4            True  1463097599000

In [ ]:
weightLogInfo.dtypes

Id                         int64
Date              datetime64[ns]
WeightKg                 float64
WeightPounds             float64
BMI                      float64
IsManualReport              bool
LogId                      int64
dtype: object

In [ ]:
weightLogInfo['Date']  = pd.to_datetime(weightLogInfo['Date'])

Next we need to formate the Date for each data

In [ ]:
dailyActivity['year'] = pd.Series()

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:1: DeprecationWarning: The default dtype for empty Series will be 'object' instead of 'float64' in a future version. Specify a dtype explicitly to silence this warning.
  """Entry point for launching an IPython kernel.


In [ ]:
dailyActivity['date'] = pd.Series()
dailyActivity['month'] = pd.Series()

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:1: DeprecationWarning: The default dtype for empty Series will be 'object' instead of 'float64' in a future version. Specify a dtype explicitly to silence this warning.
  """Entry point for launching an IPython kernel.
/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:2: DeprecationWarning: The default dtype for empty Series will be 'object' instead of 'float64' in a future version. Specify a dtype explicitly to silence this warning.
  


In [ ]:
dailyActivity['year'] = dailyActivity['ActivityDate'].dt.year

In [ ]:
dailyActivity['date'] = dailyActivity['ActivityDate'].dt.day
dailyActivity['month'] = dailyActivity['ActivityDate'].dt.month

In [ ]:
dailyActivity

Id ActivityDate  TotalSteps  TotalDistance  TrackerDistance  \
0    1503960366   2016-04-12       13162       8.500000         8.500000   
1    1503960366   2016-04-13       10735       6.970000         6.970000   
2    1503960366   2016-04-14       10460       6.740000         6.740000   
3    1503960366   2016-04-15        9762       6.280000         6.280000   
4    1503960366   2016-04-16       12669       8.160000         8.160000   
..          ...          ...         ...            ...              ...   
935  8877689391   2016-05-08       10686       8.110000         8.110000   
936  8877689391   2016-05-09       20226      18.250000        18.250000   
937  8877689391   2016-05-10       10733       8.150000         8.150000   
938  8877689391   2016-05-11       21420      19.559999        19.559999   
939  8877689391   2016-05-12        8064       6.120000         6.120000   

     VeryActiveDistance  ModeratelyActiveDistance  LightActiveDistance  \
0                  1.88                      0.55                 6.06   
1                  1.57                      0.69                 4.71   
2                  2.44                      0.40                 3.91   
3                  2.14                      1.26                 2.83   
4                  2.71                      0.41                 5.04   
..                  ...                       ...                  ...   
935                1.08                      0.20                 6.80   
936               11.10                      0.80                 6.24   
937                1.35                      0.46                 6.28   
938               13.22                      0.41                 5.89   
939                1.82                      0.04                 4.25   

     SedentaryActiveDistance  VeryActiveMinutes  FairlyActiveMinutes  \
0                       0.00                 25                   13   
1                       0.00                 21                   19   
2                       0.00                 30                   11   
3                       0.00                 29                   34   
4                       0.00                 36                   10   
..                       ...                ...                  ...   
935                     0.00                 17                    4   
936                     0.05                 73                   19   
937                     0.00                 18                   11   
938                     0.00                 88                   12   
939                     0.00                 23                    1   

     LightlyActiveMinutes  SedentaryMinutes  Calories  year        date  month  
0                     328               728      1985  2016  2016-04-12      4  
1                     217               776      1797  2016  2016-04-13      4  
2                     181              1218      1776  2016  2016-04-14      4  
3                     209               726      1745  2016  2016-04-15      4  
4                     221               773      1863  2016  2016-04-16      4  
..                    ...               ...       ...   ...         ...    ...  
935                   245              1174      2847  2016  2016-05-08      5  
936                   217              1131      3710  2016  2016-05-09      5  
937                   224              1187      2832  2016  2016-05-10      5  
938                   213              1127      3832  2016  2016-05-11      5  
939                   137               770      1849  2016  2016-05-12      5  

[940 rows x 17 columns]

In [ ]:
dailySteps['year'] = pd.Series()
dailySteps['date'] = pd.Series()
dailySteps['month'] = pd.Series()

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:1: DeprecationWarning: The default dtype for empty Series will be 'object' instead of 'float64' in a future version. Specify a dtype explicitly to silence this warning.
  """Entry point for launching an IPython kernel.
/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:2: DeprecationWarning: The default dtype for empty Series will be 'object' instead of 'float64' in a future version. Specify a dtype explicitly to silence this warning.
  
/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:3: DeprecationWarning: The default dtype for empty Series will be 'object' instead of 'float64' in a future version. Specify a dtype explicitly to silence this warning.
  This is separate from the ipykernel package so we can avoid doing imports until


In [ ]:
dailySteps['year'] = dailySteps['ActivityDay'].dt.year
dailySteps['date'] = dailySteps['ActivityDay'].dt.day
dailySteps['month'] = dailySteps['ActivityDay'].dt.month

In [ ]:
dailySteps

Id ActivityDay  StepTotal  year  date  month
0    1503960366  2016-04-12      13162  2016    12      4
1    1503960366  2016-04-13      10735  2016    13      4
2    1503960366  2016-04-14      10460  2016    14      4
3    1503960366  2016-04-15       9762  2016    15      4
4    1503960366  2016-04-16      12669  2016    16      4
..          ...         ...        ...   ...   ...    ...
935  8877689391  2016-05-08      10686  2016     8      5
936  8877689391  2016-05-09      20226  2016     9      5
937  8877689391  2016-05-10      10733  2016    10      5
938  8877689391  2016-05-11      21420  2016    11      5
939  8877689391  2016-05-12       8064  2016    12      5

[940 rows x 6 columns]

In [ ]:
sleepDay['year'] = pd.Series()
sleepDay['date'] = pd.Series()
sleepDay['month'] = pd.Series()

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:1: DeprecationWarning: The default dtype for empty Series will be 'object' instead of 'float64' in a future version. Specify a dtype explicitly to silence this warning.
  """Entry point for launching an IPython kernel.
/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:2: DeprecationWarning: The default dtype for empty Series will be 'object' instead of 'float64' in a future version. Specify a dtype explicitly to silence this warning.
  
/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:3: DeprecationWarning: The default dtype for empty Series will be 'object' instead of 'float64' in a future version. Specify a dtype explicitly to silence this warning.
  This is separate from the ipykernel package so we can avoid doing imports until


In [ ]:
sleepDay['year'] = sleepDay['SleepDay'].dt.year
sleepDay['date'] = sleepDay['SleepDay'].dt.day
sleepDay['month'] = sleepDay['SleepDay'].dt.month

In [ ]:
sleepDay

Id   SleepDay  TotalSleepRecords  TotalMinutesAsleep  \
0    1503960366 2016-04-12                  1                 327   
1    1503960366 2016-04-13                  2                 384   
2    1503960366 2016-04-15                  1                 412   
3    1503960366 2016-04-16                  2                 340   
4    1503960366 2016-04-17                  1                 700   
..          ...        ...                ...                 ...   
408  8792009665 2016-04-30                  1                 343   
409  8792009665 2016-05-01                  1                 503   
410  8792009665 2016-05-02                  1                 415   
411  8792009665 2016-05-03                  1                 516   
412  8792009665 2016-05-04                  1                 439   

     TotalTimeInBed  year  date  month  
0               346  2016    12      4  
1               407  2016    13      4  
2               442  2016    15      4  
3               367  2016    16      4  
4               712  2016    17      4  
..              ...   ...   ...    ...  
408             360  2016    30      4  
409             527  2016     1      5  
410             423  2016     2      5  
411             545  2016     3      5  
412             463  2016     4      5  

[413 rows x 8 columns]

In [ ]:
weightLogInfo['year'] = weightLogInfo['Date'].dt.year
weightLogInfo['date'] = weightLogInfo['Date'].dt.day
weightLogInfo['month'] = weightLogInfo['Date'].dt.month

In [ ]:
weightLogInfo

Id                Date    WeightKg  WeightPounds        BMI  \
0   1503960366 2016-05-02 23:59:59   52.599998    115.963147  22.650000   
1   1503960366 2016-05-03 23:59:59   52.599998    115.963147  22.650000   
2   1927972279 2016-04-13 01:08:52  133.500000    294.317120  47.540001   
3   2873212765 2016-04-21 23:59:59   56.700001    125.002104  21.450001   
4   2873212765 2016-05-12 23:59:59   57.299999    126.324875  21.690001   
..         ...                 ...         ...           ...        ...   
62  8877689391 2016-05-06 06:43:35   85.000000    187.392923  25.440001   
63  8877689391 2016-05-08 07:35:53   85.400002    188.274775  25.559999   
64  8877689391 2016-05-09 06:39:44   85.500000    188.495234  25.610001   
65  8877689391 2016-05-11 06:51:47   85.400002    188.274775  25.559999   
66  8877689391 2016-05-12 06:42:53   84.000000    185.188300  25.139999   

    IsManualReport          LogId  year  date  month  
0             True  1462233599000  2016     2      5  
1             True  1462319999000  2016     3      5  
2            False  1460509732000  2016    13      4  
3             True  1461283199000  2016    21      4  
4             True  1463097599000  2016    12      5  
..             ...            ...   ...   ...    ...  
62           False  1462517015000  2016     6      5  
63           False  1462692953000  2016     8      5  
64           False  1462775984000  2016     9      5  
65           False  1462949507000  2016    11      5  
66           False  1463035373000  2016    12      5  

[67 rows x 10 columns]

## 5. Share

### Identifying Trends and Relationships

In [ ]:
dailyActivity.describe()

Id    TotalSteps  TotalDistance  TrackerDistance  \
count  9.400000e+02    940.000000     940.000000       940.000000   
mean   4.855407e+09   7637.910638       5.489702         5.475351   
std    2.424805e+09   5087.150742       3.924606         3.907276   
min    1.503960e+09      0.000000       0.000000         0.000000   
25%    2.320127e+09   3789.750000       2.620000         2.620000   
50%    4.445115e+09   7405.500000       5.245000         5.245000   
75%    6.962181e+09  10727.000000       7.712500         7.710000   
max    8.877689e+09  36019.000000      28.030001        28.030001   

       VeryActiveDistance  ModeratelyActiveDistance  LightActiveDistance  \
count          940.000000                940.000000           940.000000   
mean             1.502681                  0.567543             3.340819   
std              2.658941                  0.883580             2.040655   
min              0.000000                  0.000000             0.000000   
25%              0.000000                  0.000000             1.945000   
50%              0.210000                  0.240000             3.365000   
75%              2.052500                  0.800000             4.782500   
max             21.920000                  6.480000            10.710000   

       SedentaryActiveDistance  VeryActiveMinutes  FairlyActiveMinutes  \
count               940.000000         940.000000           940.000000   
mean                  0.001606          21.164894            13.564894   
std                   0.007346          32.844803            19.987404   
min                   0.000000           0.000000             0.000000   
25%                   0.000000           0.000000             0.000000   
50%                   0.000000           4.000000             6.000000   
75%                   0.000000          32.000000            19.000000   
max                   0.110000         210.000000           143.000000   

       LightlyActiveMinutes  SedentaryMinutes     Calories    year  \
count            940.000000        940.000000   940.000000   940.0   
mean             192.812766        991.210638  2303.609574  2016.0   
std              109.174700        301.267437   718.166862     0.0   
min                0.000000          0.000000     0.000000  2016.0   
25%              127.000000        729.750000  1828.500000  2016.0   
50%              199.000000       1057.500000  2134.000000  2016.0   
75%              264.000000       1229.500000  2793.250000  2016.0   
max              518.000000       1440.000000  4900.000000  2016.0   

             date       month  
count  940.000000  940.000000  
mean    15.787234    4.350000  
std      8.536300    0.477224  
min      1.000000    4.000000  
25%      9.000000    4.000000  
50%     16.000000    4.000000  
75%     23.000000    5.000000  
max     30.000000    5.000000

1. The average of Steps for 33 Users is 7637. This is relatively low, based on the Health Line, healthy adults can take anywhere between 4000 - 18000 steps in a day. If we look into the data there are numbers of users who actually took more than 10000 steps a day. This indicated that the data still consist outliers.
<br>
2. The average distance is 5. However there are no informations containing what measures dose this data use. Whether it is in km, m, or miles. 
<br>
3. Average calories for each users are 2303. This is still considered good, because the best calories for women is in 2000 and for men 2500. 

In [ ]:
dailySteps.describe()

Id     StepTotal    year        date       month
count  9.400000e+02    940.000000   940.0  940.000000  940.000000
mean   4.855407e+09   7637.910638  2016.0   15.787234    4.350000
std    2.424805e+09   5087.150742     0.0    8.536300    0.477224
min    1.503960e+09      0.000000  2016.0    1.000000    4.000000
25%    2.320127e+09   3789.750000  2016.0    9.000000    4.000000
50%    4.445115e+09   7405.500000  2016.0   16.000000    4.000000
75%    6.962181e+09  10727.000000  2016.0   23.000000    5.000000
max    8.877689e+09  36019.000000  2016.0   30.000000    5.000000

In [ ]:
sleepDay.describe()

Id  TotalSleepRecords  TotalMinutesAsleep  TotalTimeInBed  \
count  4.130000e+02         413.000000          413.000000      413.000000   
mean   5.000979e+09           1.118644          419.467312      458.639225   
std    2.060360e+09           0.345521          118.344679      127.101607   
min    1.503960e+09           1.000000           58.000000       61.000000   
25%    3.977334e+09           1.000000          361.000000      403.000000   
50%    4.702922e+09           1.000000          433.000000      463.000000   
75%    6.962181e+09           1.000000          490.000000      526.000000   
max    8.792010e+09           3.000000          796.000000      961.000000   

         year        date       month  
count   413.0  413.000000  413.000000  
mean   2016.0   15.777240    4.358354  
std       0.0    8.704634    0.480098  
min    2016.0    1.000000    4.000000  
25%    2016.0    8.000000    4.000000  
50%    2016.0   16.000000    4.000000  
75%    2016.0   23.000000    5.000000  
max    2016.0   30.000000    5.000000

The average sleep minute for each users are 419 minutes. This is still in a good range. If converted into hours, it indicates the average sleep minutes for users are 7 hours. 

In [ ]:
weightLogInfo.describe()

Id    WeightKg  WeightPounds        BMI         LogId  \
count  6.700000e+01   67.000000     67.000000  67.000000  6.700000e+01   
mean   7.009282e+09   72.035821    158.811801  25.185224  1.461772e+12   
std    1.950322e+09   13.923206     30.695415   3.066963  7.829948e+08   
min    1.503960e+09   52.599998    115.963147  21.450001  1.460444e+12   
25%    6.962181e+09   61.400002    135.363832  23.959999  1.461079e+12   
50%    6.962181e+09   62.500000    137.788914  24.389999  1.461802e+12   
75%    8.877689e+09   85.049999    187.503152  25.559999  1.462375e+12   
max    8.877689e+09  133.500000    294.317120  47.540001  1.463098e+12   

         year       date      month  
count    67.0  67.000000  67.000000  
mean   2016.0  14.835821   4.402985  
std       0.0   8.699699   0.494200  
min    2016.0   1.000000   4.000000  
25%    2016.0   8.000000   4.000000  
50%    2016.0  14.000000   4.000000  
75%    2016.0  21.500000   5.000000  
max    2016.0  30.000000   5.000000

1. The average weight for 8 users are 72 and BMI is 25. However we can't really indicate anything as there is no ither demografic information such as age and height. As for the BMI, the respondents can be considered overweight.